src/System.IO/tests/BinaryWriter/BinaryWriterTests.cs

```diff
@@ -0,0 +1,345 @@
+            mstr = new MemoryStream();
+            dw2 = new BinaryWriter(mstr);
+            dw2.Write("0123456789".ToCharArray());
+            lReturn = dw2.Seek(11, SeekOrigin.Begin);  //This wont throw any exception now.
+
+            Assert.Equal(11, mstr.Position);
+
+            dw2.Dispose();
+            mstr.Dispose();
+
+            // [] Seek to the end
+            mstr = new MemoryStream();
+            dw2 = new BinaryWriter(mstr);
+            dw2.Write("0123456789".ToCharArray());
+            lReturn = dw2.Seek(10, SeekOrigin.Begin);
+
+            Assert.Equal(10, lReturn);
+
+            dw2.Write("ll".ToCharArray());
+            bArr = mstr.ToArray();
+            sb = new StringBuilder();
+            for (int i = 0; i < bArr.Length; i++)
+                sb.Append((Char)bArr[i]);
+
+            Assert.Equal("0123456789ll", sb.ToString());
+
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        [Theory]
+        [InlineData(-1)]
+        [InlineData(-2)]
+        [InlineData(-10000)]
+        [InlineData(Int32.MinValue)]
+        public static void BinaryWriter_SeekTests_NegativeOffset(int invalidValue)
+        {
+            // [] ArgumentOutOfRangeException if offset is negative
+            using (MemoryStream memStream = new MemoryStream())
+            using (BinaryWriter writer = new BinaryWriter(memStream))
+            {
+                writer.Write("Hello, this is my string".ToCharArray());
+                Assert.Throws<IOException>(() => writer.Seek(invalidValue, SeekOrigin.Begin));
+            }
+        }
+
+        [Fact]
+        public static void BinaryWriter_SeekTests_InvalidSeekOrigin()
+        {
+            // [] ArgumentException for invalid seekOrigin
+            using (MemoryStream memStream = new MemoryStream())
+            using (BinaryWriter writer = new BinaryWriter(memStream))
+            {
+                writer.Write("012345789".ToCharArray());
+
+                Assert.Throws<ArgumentException>(() => 
+                {
+                    writer.Seek(3, ~SeekOrigin.Begin);
+                });
+            }
+        }
+
+        [Fact]
+        public static void BinaryWriter_BaseStreamTests()
+        {
+            // [] Get the base stream for MemoryStream
+            using (MemoryStream ms2 = new MemoryStream())
+            using (BinaryWriter sr2 = new BinaryWriter(ms2))
+            {
+                Assert.Same(ms2, sr2.BaseStream);
+            }
+        }
+
+        [Fact]
+        public static void BinaryWriter_FlushTests()
+        {
+            // [] Check that flush updates the underlying stream
+            using (MemoryStream memstr2 = new MemoryStream())
+            using (BinaryWriter bw2 = new BinaryWriter(memstr2))
+            {
+                string str = "HelloWorld";
+                int expectedLength = str.Length + 1; // 1 for 7-bit encoded length
+                bw2.Write("HelloWorld");
```

Comment:
```bw2.Write(str);``` would be better if we're gonna use a string allocation for "HelloWorld" anyways.